core/src/main/scala/org/apache/spark/util/logging/DriverLogger.scala

```diff
@@ -219,6 +223,11 @@ private[spark] object DriverLogger extends Logging {
             logError("Could not add driver logger", e)
             None
         }
+      } else if (localDriverLogEnabled) {
+        // Driver Logger is started only for Spark Driver Log UI Tab
+        new DriverLogger(conf)
+        // Return None because we don't need DFS-related logic in SparkContext and DfsAsyncWriter
```

Comment:
Hmm, this only wants to do `addLogAppender` but doesn't need the dfs sync logic, right?

Reply:
I'm wondering if we need to call `stop`? Although for local driver log case, it just needs to remove appender without stopping sync writer.

Reply:
> Hmm, this only wants to do `addLogAppender` but doesn't need the dfs sync logic, right?

Yes, right.

> I'm wondering if we need to call stop? Although for local driver log case, it just needs to remove appender without stopping sync writer.

We don't need to call `stop` because it's used when `SparkContext` is already stopped.

https://github.com/apache/spark/blob/8d4ca0aa580f9622eb6f74bb99ae8a3e4e9667d5/core/src/main/scala/org/apache/spark/SparkContext.scala#L2257

Reply:
BTW, this is a live UI. So, when the driver JVM stopped, the log is supposed to be lost. In addition, when `DRIVER_LOG_PERSISTTODFS` is set, it will follow the original path including invoking `stop` and `sync`. However, `DRIVER_LOG_PERSISTTODFS` is a separate logic which we need to investigate if it works as designed.